### ISSUE TITLE
[Proposal] BackgroundColorTo Animation Extension

### ISSUE THREAD

**Issue by brminnick** [CommunityToolkit/Maui#88]
# ColorTo

* [x] Proposed
* [x] Prototype
* [x] Implementation
  * [x] iOS Support
  * [x] Android Support
  * [x] macOS Support
  * [x] Windows Support
* [x] Unit Tests
* [x] Sample
* [x] Documentation

## Summary
[summary]: #summary

Animation to change the Color of a VisualElement

## Detailed Design
[design]: #detailed-design

### ColorAnmiationExtensions.shared.cs

```cs
public static Task<bool> BackgroundColorTo(this VisualElement element, Color color, uint length = 250u, Easing? easing = null)
{
  ArgumentNullException.ThrowIfNull(element);
  
  var animationCompletionSource = new TaskCompletionSource<bool>();
  
  new Animation
  {
    { 0, 1, new Animation(v => element.BackgroundColor = new Color(v, element.BackgroundColor.G, element.BackgroundColor.B, element.BackgroundColor.A), element.BackgroundColor.R, color.R) },
    { 0, 1, new Animation(v => element.BackgroundColor = new Color(element.BackgroundColor.R, v, element.BackgroundColor.B, element.BackgroundColor.A), element.BackgroundColor.G, color.G) },
    { 0, 1, new Animation(v => element.BackgroundColor = new Color(element.BackgroundColor.R, element.BackgroundColor.G, v, element.BackgroundColor.A), element.BackgroundColor.B, color.B) },
    { 0, 1, new Animation(v => element.BackgroundColor = new Color(element.BackgroundColor.R, element.BackgroundColor.G, element.BackgroundColor.B, v), element.BackgroundColor.A, color.A) },
  }.Commit(element, nameof(ColorTo), 16, length, easing, (d, b) => animationCompletionSource.SetResult(true));
  
  return animationCompletionSource.Task;
}
```

## Usage Syntax
[usage]: #usage-syntax

### XAML Usage

N/A

### C# Usage

```cs
protected override async void OnAppearing()
{
  await myLabel.BackgroundColorTo(Colors.Red);
}
```

**Comment by PieEatingNinjas** [comment]
Hi @brminnick. If nobody is working on this, I want to give it a try. Haven’t done anything on this project yet, this one seems to be a good one to start with. 🙂

**Comment by brminnick** [comment]
Thanks @PieEatingNinjas! Assigned ✅

**Comment by PieEatingNinjas** [comment]
Hi @brminnick !
I've implemented the Extension method and added a sample to the sample app.
Tested on Android emulator, Windows 11 and physical iOS 15.1 device.

There seems to be a bug on Windows on some controls as their BackgroundColor doesn't get updated. I've filed a bug for that on the [MAUI repo](https://github.com/dotnet/maui/issues/3506). 
I'm not sure if/how this can be unit tested properly. If you have any ideas about this, let me know.
PR incoming! 🙂
